custom_components/tahoma/__init__.py

```diff
@@ -6,21 +6,71 @@
 
 from pyhoma.client import TahomaClient
 from pyhoma.exceptions import BadCredentialsException, TooManyRequestsException
+import voluptuous as vol
 
+from homeassistant import config_entries
 from homeassistant.components.scene import DOMAIN as SCENE
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.const import (
+    CONF_EXCLUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
 
 from .const import DOMAIN, TAHOMA_TYPES
 from .coordinator import TahomaDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 DEFAULT_UPDATE_INTERVAL = timedelta(seconds=10)
 
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                cv.deprecated(CONF_EXCLUDE, default=[]): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
 
 async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the TaHoma component."""
+    conf = config.get(DOMAIN)
+
+    if conf is None:
+        return True
+
+    if CONF_USERNAME not in conf or CONF_PASSWORD not in conf:
```

Comment:
Are you sure it can occur? Voluptuous should take of this for us.

Reply:
Good one, not sure. I will need to test and update.

Reply:
```
2020-08-02 22:22:48 ERROR (MainThread) [homeassistant.config] Invalid config for [tahoma]: string value is None for dictionary value @ data['tahoma']['password']. Got None. (See /Users/mick/.homeassistant/configuration.yaml, line 38). Please check the docs at https://www.home-assistant.io/integrations/tahoma
2020-08-02 22:22:48 ERROR (MainThread) [homeassistant.setup] Setup failed for tahoma: Invalid config.
```

A wrong config is indeed validated by voluptuous.